sipa/blueprints/generic.py

```diff
@@ -162,7 +165,18 @@ def logout():
                 extra={'tags': {'user': current_user.uid}})
     logout_user()
     flash(gettext("Abmeldung erfolgreich!"), 'success')
-    return redirect(url_for('.index'))
+
+    resp = redirect(url_for('.index'))
+
+    cookie_name = current_app.config.get('REMEMBER_COOKIE_NAME', COOKIE_NAME)
+    if cookie_name in request.cookies:
+        resp.set_cookie(cookie_name, "", 0)
+
+    current_app.login_manager._clear_cookie(resp)
+    resp.set_cookie("session", "")
+    session.clear()
+
+    return resp
```

Comment:
I believe @sgeisler is attempting to do a “stricter” cleanup by deleting the whole session.

`logout_user` only pops `user_id` and `_fresh` from the session, plus resets the `remember_me` cookie. See [the source](https://github.com/maxcountryman/flask-login/blob/14e24935baf92f79233cb342dbc93c5ac546439f/flask_login/utils.py#L182).